Added `org.eclipse.jetty.toolchain:jetty-jakarta-servlet-api` to `JavaxServletApiRule`

DreierF opened the pull request:
For that one I'm not really sure whether the behavior is the desired one.
The `org.eclipse.jetty.toolchain:jetty-jakarta-servlet-api` dependency seems to be a bundle that contains both the `javax.servlet:servlet-api`, `jakarta.servlet:jakarta.servlet-api`, additional `module-info.java` and xml files.

jjohannes commented:
I can't see any `javax` classes in https://repo1.maven.org/maven2/org/eclipse/jetty/toolchain/jetty-jakarta-servlet-api/5.0.0/jetty-jakarta-servlet-api-5.0.0.jar

But the _jakarta_ variant is in there.

In my mind, this has to move to `JakartaServletApiRule`, or do I miss something @DreierF?

If not, would you mind updating this PR? If you rebase `main`, the `samples/sample-all` test will now tell you if there are duplicated classes when you add a new dependency (and the rules aren't right yet).

DreierF commented:
Thanks for taking a look!
You are right! I was mislead by the fact that the jar contains some non-class files in the javax folder, but it actually only contains jakarta classes.
I moved the artifact to `JakartaServletApiRule` and rebased on lastest main.
